core/src/validator.rs

```diff
@@ -414,6 +414,19 @@ impl Validator {
 
         let accounts_package_channel = channel();
 
+        let accounts_update_notifier = if accountsdb_plugin_service.is_some() {
+            accountsdb_plugin_service
+                .as_ref()
+                .unwrap()
+                .get_accounts_update_notifier()
+        } else {
+            None
+        };
```

Comment:
I think you can bundle this change into the one-liner you had before by using `and_then()` instead of `map()`